### DISCUSSION TITLE
Changelog

### DISCUSSION THREAD

**Discussion by tefkah** [org-roam/org-roam-ui#30] [Feature request]
Until we have found a better solution, I (or @kirillrogovoy ) will post updates here, a new post for each update. You can subscribe to this thread in case you want to keep up to date with the latest and greatest of org-roam-ui

**Comment by tefkah** [comment]
Aug 2: Tag coloring and filtering 

https://user-images.githubusercontent.com/21983833/127928395-cb331ca6-4625-41d4-ba27-5de2be647eb9.mp4

**Comment by tefkah** [comment]
Aug 3: Cite and Ref link support thanks to @blester125

**Comment by tefkah** [comment]
Aug 6: Lot's of new things! I will be making a video about it tomorrow, but in short:
UI:
- Longer tweaks
- Fade nodes/links on zoom
- Title now below node
- Better defaults
- You can now set how much the number of links contributes to a nodes size

Graph:
- You can now see "dead" nodes, nodes which do not lead anywhere
- Big one: updates to the graph now merge gracefully (as much as they can) and don't refresh the entire graph!
- Context menu! Open local graphs, create notes for nodes without files, and delete nodes right from org-roam-ui! Much more to come. You should see an "open in zotero" one as well, which does not work yet. Node creation either opens the node with `orb-edit-note` if it's a citation node and you have org roam bibtex installed, or just opens a plain node template. FILELESS nodes have a tag called "bad" which you can use to color them, it will automaticlaly be set once you set the filter.
Let me know what kind of features you would like to see next!

![image](https://user-images.githubusercontent.com/21983833/128438489-53243bc4-6fb2-469a-b3fe-80605906564e.png)

![image](https://user-images.githubusercontent.com/21983833/128438518-9e5ed8af-c260-4ccd-8828-57b4c2c5f979.png)

![image](https://user-images.githubusercontent.com/21983833/128438532-3546623a-f852-467b-a19a-32347030a0bb.png)

![image](https://user-images.githubusercontent.com/21983833/128438570-8e0c8a8e-9f2c-4fdd-b7b1-fce968890075.png)

**Comment by tefkah** [comment]
Aug 9: Entering and exiting local mode is now much smoother. This is something we will continue to polish over time, and we would really appreciate the feedback on this implementation and the local mode in general!

https://user-images.githubusercontent.com/21983833/128647540-b77ad1d0-d0b7-48c3-a966-f17f3f37ef59.mp4

**Comment by tefkah** [comment]
Aug 16: The "link to heading node" behavior is now available in the latest build! You can toggle between linking the child nodes to the next highest heading node, linking them to the parent file, or not link them at all.

New:
![image](https://user-images.githubusercontent.com/21983833/129583838-344b2b59-27b2-4b70-bd2c-4bbf128ff0f2.png)

Old:
![image](https://user-images.githubusercontent.com/21983833/129583973-a8e17cbd-7779-442e-a5a5-6a022deb9585.png)

**Reply by tefkah** [reply]
Note: there is currently a bug with the highlighting, will fix asap.

**Reply by midas0441** [reply]
awesome!

**Reply by tefkah** [reply]
This bug has been fixed as of https://github.com/org-roam/org-roam-ui/commit/2f6545337d832c766feaae1854b936bb3d1d9557

**Comment by tefkah** [comment]
Aug 29: Org-roam changed it's db structure and citation aggregation logic to support the new `org-cite` feature in Org-mode. Org-roam-ui can now also detect such citations.

**Comment by tefkah** [comment]
Sept 13: Added the option to keep node size relatively invariant across zoom levels.

The default strength of this effect is approx 1/(Z*Z^1/3), but this can be varied with yet another slider. To regain the default behavior, set it to 0.
Before:
![image](https://user-images.githubusercontent.com/21983833/133093586-fd34c775-1bf9-4ea4-9db5-f95bd83f4574.png)
![image](https://user-images.githubusercontent.com/21983833/133093648-13639511-40b8-4eb9-95b1-35947218e541.png)

After:
![image](https://user-images.githubusercontent.com/21983833/133094207-bbca9464-a259-48d4-a773-b074fad125c9.png)
![image](https://user-images.githubusercontent.com/21983833/133093877-d3460ba7-5a91-486b-a625-a3d5113691bd.png)

**Reply by tefkah** [reply]
If anyone has an idea for how to *really* keep the node size consistent across scales let me know! This works well enough, but could be better.

**Reply by maikol-solis** [reply]
Where can we activate this option? Thanks

**Reply by tefkah** [reply]
Under "Visual > Nodes & Links"!

**Comment by tefkah** [comment]
Oct 6: In ui theme switching. It was already possible to change org-roam-ui to match your Emacs theme, but now you can find a ton of theme presets (based on doom-themes) in the Visuals menu!


https://user-images.githubusercontent.com/21983833/136130896-0895ff67-ea3a-4432-a050-b1a29356267b.mov

**Comment by tefkah** [comment]
Oct 6: New filter for dailies.

https://user-images.githubusercontent.com/21983833/136193355-18b06e9a-42c1-442c-972c-a883099b6e55.mov

**Comment by tefkah** [comment]
Oct 11: File previewing is live!

https://user-images.githubusercontent.com/21983833/136845036-a6bd4b32-d78c-4bcf-aa8a-0e37c69cbbe8.mp4

**Comment by tefkah** [comment]
October 15: Changing the number of neighbors shown in local mode

**Comment by tefkah** [comment]
Oct 16: 
- Dynamic node label appearance. Labels now (by default, can be disabled) appear later the lesser the number of links they have. The strength of this effect (default: square of the zoom level) and the max number of links that count (default: 8, after this the degree of the nodes is treated to same) can be configured from the settings.
- The labels also behave a bit more consistent in general, keeping the same distance from the nodes.

**Comment by tefkah** [comment]
Oct 17: Color by community

https://user-images.githubusercontent.com/21983833/137604335-b98e123e-9de4-4ad5-b546-66d3b2574729.mov

This is one of the coloring algorithms I want to provide, which colors the nodes not by the amount of links they have (their "degree") but by the "community" the node is a part of (modularity). Atm it recomputes it when you change the filter (not the local/global mode) because often times you don't want a lot of the nodes in your graph to show. This does mean that the communities are largely unstable, in the sense that you won't be able to associate "green" with "notes on topology" or something, but I'm not sure that that's feasible in the first place.

I decided to put this out here now to see whether people like/use it, so let me know what you think! I personally think it's really useful as it allows you see at a glance which nodes "belong" together, and gives you ideas on how to connect disparate communities of nodes.

**Reply by tefkah** [reply]
I'd like to provide some sort of setting for the "granularity" of the clusters, but the package I'm currently using doesn't really provide that option, and the package that I want to use (ngraph) is very cumbersome and the default result is too granular for my liking. I'll probably move to the latter anyway later, since they provide a lot of different algorithms (shortest path, centrality measures, CW instead of Louvain communities) which might be interesting later.

**Comment by tefkah** [comment]
Nov 2: Outline view and collapsible headers

Kind of rudimentary atm, let me know what you think!

https://user-images.githubusercontent.com/21983833/139960383-ca6e95fc-f3e6-4658-9959-8adb78cef9f8.mov

**Comment by tefkah** [comment]
Nov 13:
Support for Markdown formatted notes via [md-roam](https://github.com/nobiot/md-roam), with wiki-links!
<img width="1192" alt="Screenshot 2021-11-13 at 13 16 02" src="https://user-images.githubusercontent.com/21983833/141643455-76021ac3-76a6-46c8-83b9-cb0c537673bf.png">

**Comment by tefkah** [comment]
You can now (finally!) just open a non-top level node in the file viewer! Thanks @egh!

**Comment by tefkah** [comment]
Add ability to add/remove nodes from the local graph with `org-roam-ui-add-to-local-graph` and `org-roam-ui-remove-from-local-graph`
